tests/js/server/aql/aql-index-choice.js

```diff
@@ -0,0 +1,513 @@
+      db[cn].ensureIndex({ type: "persistent", fields: ["uid"] });
+      
+      [
+        `FOR doc IN ${cn} FILTER doc.uid == 1234 RETURN doc.uid`,
+        `FOR doc IN ${cn} FILTER doc.uid > 1234 RETURN doc.uid`,
+        `FOR doc IN ${cn} FILTER doc.uid >= 1234 RETURN doc.uid`
+      ].forEach((q) => {
+        let nodes = AQL_EXPLAIN(q).plan.nodes;
+        assertEqual(1, nodes.filter((n) => n.type === 'IndexNode').length);
+        assertEqual(0, nodes.filter((n) => n.type === 'SortNode').length);
+        let indexNode = nodes.filter((n) => n.type === 'IndexNode')[0];
+        assertEqual(["uid"], indexNode.projections);
+        assertTrue(indexNode.indexCoversProjections);
+        assertEqual(1, indexNode.indexes.length);
+        let index = indexNode.indexes[0];
+        assertEqual("persistent", index.type);
+        assertEqual(["uid"], index.fields);
+      });
+    },
+    
+    testMultipleIndexesSingleFilter: function() {
+      db[cn].ensureIndex({ type: "persistent", fields: ["uid"] });
+      db[cn].ensureIndex({ type: "persistent", fields: ["uid", "dt"] });
+      
+      [
+        `FOR doc IN ${cn} FILTER doc.uid == 1234 RETURN doc`,
+        `FOR doc IN ${cn} FILTER doc.uid > 1234 RETURN doc`,
+        `FOR doc IN ${cn} FILTER doc.uid >= 1234 RETURN doc`
+      ].forEach((q) => {
+        let nodes = AQL_EXPLAIN(q).plan.nodes;
+        assertEqual(1, nodes.filter((n) => n.type === 'IndexNode').length);
+        assertEqual(0, nodes.filter((n) => n.type === 'SortNode').length);
+        let indexNode = nodes.filter((n) => n.type === 'IndexNode')[0];
+        assertEqual([], indexNode.projections);
+        assertFalse(indexNode.indexCoversProjections);
+        assertEqual(1, indexNode.indexes.length);
+        let index = indexNode.indexes[0];
+        assertEqual("persistent", index.type);
+        assertEqual(["uid", "dt"], index.fields);
+      });
+    },
+    
+    testMultipleIndexesSingleFilterAndProjection: function() {
+      db[cn].ensureIndex({ type: "persistent", fields: ["uid"] });
+      db[cn].ensureIndex({ type: "persistent", fields: ["uid", "dt"] });
+      
+      [
+        `FOR doc IN ${cn} FILTER doc.uid == 1234 RETURN doc.uid`,
+        `FOR doc IN ${cn} FILTER doc.uid > 1234 RETURN doc.uid`,
+        `FOR doc IN ${cn} FILTER doc.uid >= 1234 RETURN doc.uid`
+      ].forEach((q) => {
+        let nodes = AQL_EXPLAIN(q).plan.nodes;
+        assertEqual(1, nodes.filter((n) => n.type === 'IndexNode').length);
+        assertEqual(0, nodes.filter((n) => n.type === 'SortNode').length);
+        let indexNode = nodes.filter((n) => n.type === 'IndexNode')[0];
+        assertEqual(["uid"], indexNode.projections);
+        assertTrue(indexNode.indexCoversProjections);
+        assertEqual(1, indexNode.indexes.length);
+        let index = indexNode.indexes[0];
+        assertEqual("persistent", index.type);
+        assertEqual(["uid", "dt"], index.fields);
```

Comment:
Why would be pick the index with two fields in this scenario? `dt` is not used, neither in a filter nor in a projection.